tests/admin_views/admin.py

```diff
@@ -156,6 +156,10 @@ def has_change_permission(self, request, obj=None):
         """ Only allow changing objects with even id number """
         return request.user.is_staff and (obj is not None) and (obj.id % 2 == 0)
 
+    def has_view_permission(self, request, obj=None):
```

Comment:
As per comments on `test_change_view` and `test_history_view`, I don't think this is really behaving in the expected way, or really testing the desired behaviour. 

(TODO: pull out the row-level permissions parts of those tests in a separate commit, and then this can be skipped.)

Reply:
You mean a separate commit but still in the PR ? (or take it out of the PR altogether and leave a TODO note in the code ?)